src/cr/cube/dimension.py

```diff
@@ -381,6 +381,16 @@ def prune(self):
             return True
         return False
 
+    @lazyproperty
+    def selected_categories(self):
+        """List of selected categories specified for this dimension."""
+        selected_categories = self._dimension_dict["references"].get(
+            "selected_categories"
+        )
+        if not selected_categories or selected_categories == [{}]:
```

Comment:
Why is `[{}]` a legitimate value for selected_categories? If `dimension.references.selected_categories` is an array of categories, `[]` is a valid value, but `{}` is not a valid category and should be filtered out by the producer, not left to clients to deal with.